src/components/WebexInMeeting/WebexInMeeting.jsx

```diff
@@ -17,29 +26,63 @@ import {TABLET, DESKTOP, DESKTOP_LARGE} from '../breakpoints';
  * @param {string} props.layout  Layout to apply on remote video
  * @param {string} props.meetingID  ID of the meeting for which to show media
  * @param {object} props.style  Custom style to apply
+ * @param {object} props.setEscapedAuthentication Set true if user escapes authentication
  * @returns {object} JSX of the component
  */
 export default function WebexInMeeting({
-  className, layout, meetingID, style,
+  className, layout, meetingID, style, setEscapedAuthentication,
 }) {
-  const {remoteShare, localShare} = useMeeting(meetingID);
+  const adapter = useContext(AdapterContext);
+  const {
+    remoteShare,
+    localShare,
+    passwordRequired,
+    state,
+  } = useMeeting(meetingID);
   const meetingRef = useRef();
   const {width, height} = useElementDimensions(meetingRef);
   const [maxWidth, setMaxWidth] = useState('none');
+  const [authModal, setAuthModal] = useState('guest');
   const localMediaType = localShare?.stream ? 'screen' : 'video';
   const [cssClasses, sc] = webexComponentClasses('in-meeting', className, {
     tablet: width >= TABLET && width < DESKTOP,
     desktop: width >= DESKTOP && width < DESKTOP_LARGE,
     'desktop-xl': width >= DESKTOP_LARGE,
     'remote-sharing': remoteShare !== null,
   });
+  const {NOT_JOINED} = MeetingState;
 
   useEffect(() => {
     setMaxWidth(height ? (height * 16) / 9 : 'none');
   }, [height]);
 
+  const onCloseHandler = () => {
+    adapter.meetingsAdapter.clearPasswordRequiredFlag(meetingID);
+    setEscapedAuthentication(true);
+  };
+
   return (
     <div ref={meetingRef} className={cssClasses} style={style}>
+      {passwordRequired && state === NOT_JOINED && (
+      <Modal
+        onClose={onCloseHandler}
+        otherClassName={[sc('authentication')]}
+        onBack={authModal === 'host' && (() => setAuthModal('guest'))}
```

Comment:
can we keep variable which will hold `isHostAuthModal` `isGuestAuthModal` and then use where ever it's needed?